lib/rules/no-constant-condition.js

```diff
@@ -143,11 +204,18 @@ module.exports = {
         //--------------------------------------------------------------------------
 
         return {
-            ConditionalExpression: checkConstantCondition,
-            IfStatement: checkConstantCondition,
+            ConditionalExpression: reportIfConstant,
+            IfStatement: reportIfConstant,
             WhileStatement: checkLoop,
+            "WhileStatement:exit": checkConstantConditionLoopInSet,
             DoWhileStatement: checkLoop,
-            ForStatement: checkLoop
+            "DoWhileStatement:exit": checkConstantConditionLoopInSet,
+            ForStatement: checkLoop,
+            "ForStatement > .init * > YieldExpression, ForStatement > .init": node => checkYieldInInitForLoop(node),
```

Comment:
@not-an-aardvark awesome thanks for the review. my bad - I didn't catch to look at the test part of the forloop to decide if something would run as a constant.

I updated the test cases.